packages/jest-editor-support/src/parsers/TypeScriptParser.js

```diff
@@ -9,79 +9,93 @@
  */
 
 'use strict';
+try {
+  const ts = require('typescript');
+  if (!ts) {
+    throw new Error('no typescript');
+  }
```

Comment:
Yeah, just make it so the module throws when you try to require it and typescript isn't available. Actually, you don't need to make any changes to this module. Whoever uses this with typescript (@orta) will have typescript already as a dep.